### PR TITLE
feat(login): allow anonymous access to specific views

### PR DESCRIPTION
**Pull request by andrewazores**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] Signed the last commit: `git commit --amend --signoff`
_______________________________________________

Fixes: #800

## Description of the change:
Adds a configuration property to allow anonymous access to specific in-app routes.

## Motivation for the change:
This allows the user to modify `cryostat-web` settings, and potentially interact with other future views, without logging in to the Cryostat backend or any associated auth server.
See #793 - this enables the language selector to be moved into the Settings view so the user can change the language of the auth page.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,7 +97,9 @@
     "@patternfly/react-table": "^4.112.6",
     "@reduxjs/toolkit": "^1.9.0",
     "@types/lodash": "^4.14.191",
+    "@types/react-router-dom": "^5.3.3",
     "i18next": "^22.4.9",
+    "i18next": "^22.4.3",
```
Comment (tthvo) [inline package.json:102]: Seem to be a duplicate as above?

File: src/app/routes.tsx [hunk 2]
```diff
@@ -215,34 +226,45 @@ export interface AppRoutesProps {}
...
-                component={component}
-                key={idx}
-                title={title}
-                isAsync={isAsync}
-              />
-            ))
-          ) : (
-            <Login />
-          )}
+          {loggedIn
```
Comment (tthvo) [inline src/app/routes.tsx:243]: Seem to have the same logic in the last `map` call. Maybe we can do:

```ts
flatten(routes)
  .filter((r) => loggedIn? r.component !== Login : r.anonymous }
  .map(...)
```

**Review by tthvo — APPROVED** [review]
Looks good to me! Worked well :D

**Review by maxcao13 — APPROVED** [review]
Works well!

**Pull request merged**